Automattic/jetpack
# Synced Newspack Blocks: Update to 4.5.2

This updates the synced Newspack Blocks from 4.0.1 to 4.5.2:
https://github.com/Automattic/newspack-blocks/releases/tag/v4.5.2

It also does the following:
* Allows one to resync the existing version of blocks.
* Adjusts the script to automate a few more tasks listed in `README.md` (e.g checks for `ENT_COMPAT`, Phan baseline updates).
* Adds a temporary `eslint.config.mjs` so we only lint the `@wordpress/i18n-text-domain` rule.
* Adds a `.phpcsignore` to the package root so the synced blocks aren't auto-formatted by the pre-commit hook.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Make sure the changelog entry is adequate.
* Verify `ENT_COMPAT` check catches as many appropriate matches as possible.
* Ensure `./bin/sync-newspack-blocks.sh --release=v4.5.2` matches the checked-in files of this branch, with one `ENT_COMPAT` exception.
